### PR TITLE
feat: add bigquery export samples

### PR DESCRIPTION
**Pull request by rafaelMurata**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vtwo.bigquery;
+
+// [START securitycenter_create_bigquery_export_v2]
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java:19]: This tag should enclose only the function `public static void createBigQueryExport(`, no?

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  private static final String BQ_DATASET_NAME = "test-dataset-id";
+  private static final String BQ_EXPORT_ID = "test-export-id";
+
+  @Test
+  public void testCreateBigQueryExport() throws IOException {
+    // Mock SecurityCenterClient.
+    SecurityCenterClient client = mock(SecurityCenterClient.class);
+    try (MockedStatic<SecurityCenterClient> clientMock = Mockito.mockStatic(
+        SecurityCenterClient.class)) {
+      clientMock.when(SecurityCenterClient::create).thenReturn(client);
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:88]: I guess these lines could be put in a @Before method, once it's used in all of the tests.

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: delete extra line here

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+      // Define test data.
+      String filter = "test-filter";
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: delete extra line

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+      // Build the parent of the request.
+      OrganizationLocationName organizationName = OrganizationLocationName.of(ORGANIZATION_ID,
+          LOCATION);
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: delete extra line

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+      // Build the BigQueryExport response.
+      BigQueryExport expectedExport = BigQueryExport.newBuilder()
+          .setDescription(
+              "Export low and medium findings if the compute resource has an IAM anomalous grant")
+          .setFilter(filter)
+          .setDataset(String.format("projects/%s/datasets/%s", PROJECT_ID, BQ_DATASET_NAME))
+          .build();
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: delete extra line

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+      // Build the CreateBigQueryExportRequest request.
+      CreateBigQueryExportRequest expectedRequest = CreateBigQueryExportRequest.newBuilder()
+          .setParent(organizationName.toString())
+          .setBigQueryExport(expectedExport)
+          .setBigQueryExportId(BQ_EXPORT_ID)
+          .build();
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:108]: delete extra line

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+      CreateBigQueryExport.createBigQueryExport(ORGANIZATION_ID, LOCATION, PROJECT_ID, filter,
+          BQ_DATASET_NAME, BQ_EXPORT_ID);
+
+      // Verify that the createBigQueryExport method was called with the expected request.
+      Mockito.verify(client).createBigQueryExport(expectedRequest);
+    }
+  }
+
+  @Test
+  public void testDeleteBigQueryExport() throws IOException {
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:122]: please do the same I asked for the first test on the tests below

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+
+      // Mock the createBigQueryExport method to return the expected response.
+      Mockito.when(client.createBigQueryExport(any())).thenReturn(expectedExport);
+
+      // Call the createBigQueryExport method.
+      CreateBigQueryExport.createBigQueryExport(ORGANIZATION_ID, LOCATION, PROJECT_ID, filter,
+          BQ_DATASET_NAME, BQ_EXPORT_ID);
+
+      // Verify that the createBigQueryExport method was called with the expected request.
+      Mockito.verify(client).createBigQueryExport(expectedRequest);
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: Please statically import `Mockito.verify()` so you can write this line as `verify(client).createBigQueryExport(expectedRequest);`

Same below!

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+
+      // Build the CreateBigQueryExportRequest request.
+      CreateBigQueryExportRequest expectedRequest = CreateBigQueryExportRequest.newBuilder()
+          .setParent(organizationName.toString())
+          .setBigQueryExport(expectedExport)
+          .setBigQueryExportId(BQ_EXPORT_ID)
+          .build();
+
+      // Mock the createBigQueryExport method to return the expected response.
+      Mockito.when(client.createBigQueryExport(any())).thenReturn(expectedExport);
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: Please statically import `Mockito.when()` so you can write this line as `when(client.createBigQueryExport(any())).thenReturn(expectedExport);`

**Review by arthurlapertosa — CHANGES_REQUESTED** [review]

File: security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java (hunk 1 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java:19]: it makes sense if we follow the same example we used in GetIamPolicies:
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/security-command-center/snippets/src/main/java/vtwo/iam/GetIamPolicies.java

but I'm worried about the change, as the same PR is already approved in the same syntax.
https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9291

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 2 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:88]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 3 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 4 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 5 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 6 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 7 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:108]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 8 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java:122]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 9 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 10 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done

File: security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java (hunk 1 above)
Comment (arthurlapertosa) [reply security-command-center/snippets/src/main/java/vtwo/bigquery/CreateBigQueryExport.java:19]: ok, no problem

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+    // Build the BigQuery export name.
+    BigQueryExportName bigQueryExportName = BigQueryExportName.of(ORGANIZATION_ID, LOCATION,
+        BQ_EXPORT_ID);
+    // Build the delete request.
+    DeleteBigQueryExportRequest bigQueryExportRequest = DeleteBigQueryExportRequest.newBuilder()
+        .setName(bigQueryExportName.toString())
+        .build();
+    // Mock the deleteBigQueryExport method to return successfully.
+    doNothing().when(client).deleteBigQueryExport(bigQueryExportRequest);
+    // Call the deleteBigQueryExport method.
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: Add line above, remember http://go/unit-testing-practices#structure

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+    DeleteBigQueryExport.deleteBigQueryExport(ORGANIZATION_ID, LOCATION, BQ_EXPORT_ID);
+    // Verify that the deleteBigQueryExport method was called with the expected request.
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: Add line above, remember http://go/unit-testing-practices#structure

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+  @AfterClass
+  public static void cleanUp() {
+    clientMock.reset();
+  }
+
+  @Test
+  public void testCreateBigQueryExport() throws IOException {
+    // Mock SecurityCenterClient.
+    SecurityCenterClient client = mock(SecurityCenterClient.class);
+    clientMock.when(SecurityCenterClient::create).thenReturn(client);
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: Can you also move these two lines to the setUp() method? (same for other methods)

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 11 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done.

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 12 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: done.

File: security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java (hunk 13 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/test/java/vtwo/BigQueryExportIT.java]: I found a way to move the two lines. The other attempts didn't work.

**Review by arthurlapertosa — APPROVED** [review]

**Pull request closed without merging**